### PR TITLE
Added code format check of `.proto` files

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
Added `check_style_proto` in the `tools` directory. It's a simple bash script
which is run in the check_format_code.yaml action.

### CODE REVIEW

**Comment by ArthurBandaryk** [comment]
Also tested `check_style_proto` script locally. Was playing with correct format, then added some format errors and made some lines exceed 80 characters. The results and `test.proto` file are on the screen below.
Screen Shot 2021-09-30 at 9 40 15 PM
Screen Shot 2021-09-30 at 9 41 04 PM
Screen Shot 2021-09-30 at 9 41 37 PM

File: .github/workflows/check_format_code.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@ jobs:
     - uses: actions/checkout@v2
     - name: Run bootstrap
       run: cd tools/ && ./bootstrap
+    - name: Install prototool
+      run: brew install prototool
```
Comment (rjhuijsman) [inline .github/workflows/check_format_code.yml:13]: Huh, I didn't know Homebrew exists on Linux too! And it's preinstalled on GitHub's Ubuntu? Cool.

File: .github/workflows/check_format_code.yml [hunk 2]
```diff
@@ -9,6 +9,10 @@ jobs:
     - uses: actions/checkout@v2
     - name: Run bootstrap
       run: cd tools/ && ./bootstrap
+    - name: Install prototool
+      run: brew install prototool  
+    - name: Check all .proto files on correct code format
```
Comment (rjhuijsman) [inline .github/workflows/check_format_code.yml:14]: English suggestion:
```suggestion
    - name: Check all .proto files for code formatting
```

File: tools/check_style_proto [hunk 3]
```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
```
Comment (rjhuijsman) [inline tools/check_style_proto:1]: Let's name this file `check_style_proto.sh` to indicate it's a shell script.

File: tools/check_style_proto [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+# Check for existence of prototool
```
Comment (rjhuijsman) [inline tools/check_style_proto:3]: All comments are full sentences, so should end with a period (`.`).

File: tools/check_style_proto [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    if [[ ${format_status} != 0 ]]; then
+      tput bold # Bold text in terminal
+      tput setaf 1 # Red font in terminal
+      printf "${file:2} file has incorrect code format!\n"
+      tput sgr0
+      printf "You can format proto files with prototool.\n"
+      tput bold # Bold text in terminal
+      printf "Command to format .proto files: "
+      tput setaf 2 # Green font in terminal
+      printf "prototool format -w file_name\n"
```
Comment (rjhuijsman) [inline tools/check_style_proto:55]: Would be helpful to fill in the real filename here.

File: tools/check_style_proto [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    # In the documentation they tell us that in order to
+    # write a lint error in the form file:line:column:message
+    # if a file is unformatted we should use the next command:
+    # prototool format [file] -l (or --lint)
+    # but in fact it only prints that Format returned a diff.
+    # Actually if the file is well formatted the command above will
+    # return 0. So I used this fact in the code below!
+    # The link to the official doc:
+    # https://github.com/uber/prototool#prototool-format
+    # Read the Command Overview section!
```
Comment (rjhuijsman) [inline tools/check_style_proto:40]: Noted. Just leave your explanation here and it'll be great:

```suggestion
    # Check if the specific file has correct code format
    #
    # In the documentation they tell us that in order to
    # write a lint error in the form file:line:column:message
    # if a file is unformatted we should use the following command:
    #   prototool format [file] -l (or --lint)
    # ... but in fact that only prints an uninformative message.
    #
    # If the file is well formatted the command above will return 0. 
    # We use this fact to print a more helpful message.
```

File: tools/check_style_proto [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      printf "You can format proto files with prototool.\n"
+      tput bold # Bold text in terminal
+      printf "Command to format .proto files: "
+      tput setaf 2 # Green font in terminal
+      printf "prototool format -w file_name\n"
+      tput sgr0 # Reset terminal.
+      exit 1
+    fi
+
+    # Check if the specific line has more than 80 characters
```
Comment (rjhuijsman) [inline tools/check_style_proto:60]:
```suggestion
    # Check if any line has more than 80 characters.
```

File: tools/check_style_proto [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        printf "error: "
+        tput sgr0 # Reset terminal.
+        tput bold
+        printf "line exceeds 80 characters\n"
+        tput sgr0 # Reset terminal.
+        printf "%s\n" "${line}"
+      fi
+      (( number++ ))
+    done < ${file}
+    unset IFS
```
Comment (rjhuijsman) [inline tools/check_style_proto:79]: Shall we move all of this logic into a separate check that we apply to _both_ `.proto` files _and_ `.cc`/`.h` files?

(That would be a separate PR, we'd just omit it here for now).

**Review by rjhuijsman — CHANGES_REQUESTED** [review]

File: tools/check_style_proto (hunk 3 above)
Comment (while-false) [reply tools/check_style_proto:1]: +1

File: tools/check_style_proto (hunk 5 above)
Comment (while-false) [reply tools/check_style_proto:55]: For `clang-format`, that would imply something like
`printf "clang-format --style=file -i \"%s\"\n" "${file}"`

File: tools/check_style_proto [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
+    printf "Checking ${file:2} ...\n"
+    prototool format -l ${file:2}
+    format_status=$(echo $?)
+
+    if [[ ${format_status} != 0 ]]; then
+      tput bold # Bold text in terminal
+      tput setaf 1 # Red font in terminal
+      printf "${file:2} file has incorrect code format!\n"
+      tput sgr0
```
Comment (while-false) [inline tools/check_style_proto:50]: I suggest `${file:2} -> ${file}` - the `./` is not that bad. ;D

File: tools/check_style_proto (hunk 8 above)
Comment (while-false) [reply tools/check_style_proto:79]: I don't understand your parenthesis?

**Review by while-false — CHANGES_REQUESTED** [review]

**Comment by rjhuijsman** [comment]
To avoid any confusion:
* This PR is the sibling of `stout-grpc`'s https://github.com/3rdparty/stout-grpc/pull/24
* I reviewed this PR before I realized that Gorm had done a review on the sibling PR, where he made some really good points.
* Gorm's review on the other PR essentially makes my review on this PR obsolete - we should ignore what I said above and work on the sibling PR, then re-make this PR to match the sibling.

To avoid further confusion, I'm going to close this PR for now. @ArthurBandaryk please feel free to send another PR once the PR in `stout-grpc` has been checked in.
